openpathsampling/sample.py

```diff
@@ -119,14 +121,15 @@ def __setitem__(self, key, value):
         self.append(value)
 
     def __eq__(self, other):
-        if len(self.samples) == len(other.samples):
-            for samp1 in self.samples:
-                if samp1 not in other.samples:
-                    return False
-
-            return True
-        else:
-            return False
+        return set(self.samples) == set(other.samples)
```

Comment:
can you add a length check to this as well? you could have duplicated samples in a sample set, then I would argue that these wouldn't be equal